File: learn_cpp/while.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

int main(int argc, char const *argv[])
{
    // while
    int i = 1;
    while (i <= 3)
    {
        cout << i << endl;
        i++;
    }

    // break: go out of loop
    int c = 2;
    while (true) {
        if (c == 10)
            break;
        
        cout << c << endl;
        c++;        
    }

    // continue: go first of loop
    int k = 1;
    while (k != 10)
    {
        if (k % 2 == 0)
        {
            k++; // if you comment this line you can see infinite loop
            continue;
        }
        cout << k << endl;
        k++;
    }

    // nested while
    // ... :)
    


    return 0;
}

File: learn_dart/while.dart
void main(List<String> args) {
  // while
  int i = 0;
  while (i <= 3) {
    print(i);
    i++;
  }

  // continue: go to first of loop
  int k = 1;
  while (k != 10) {
    if (k % 2 == 0) {
      k++; // if you comment this line you see infinite loop
      continue;
    }
    print(k);
    k++;
  }

  // break: go out of loop
  int c = 1;
  while (true) {
    if (c == 10) break;

    print(c);
    c++;
  }

  // nested whiles
  // ...
}
